src/EventStore.Client.Streams/EventStoreClient.Append.cs

```diff
@@ -255,7 +258,13 @@ await foreach (var response in call.ResponseStream.ReadAllAsync(_cancellationTok
 
 						try {
 							writeResult.TrySetResult(response.ToWriteResult());
-						} catch (Exception ex) when (ex is not RpcException) {
+						} catch (RpcException ex) when (ex.StatusCode is not StatusCode.NotFound or
```

Comment:
**QUESTION:** Why was this additional condition added for `RpcException` here?

Reply:
i think a comment would be in order to explain why its these specific statuses

... although DeadLineExceeded should probably be in this list shouldn't it, otherwise a timeout would cause the batchappender to terminate

... in fact, isn't DeadlineExceeded the only possible RpcException we could get here?